### ISSUE TITLE
nxcomp 3.6.x FTBFS on Cygwin

### ISSUE THREAD

**Issue by mikedep333** [ArcticaProject/nx-libs#394]
Using latest 32-bit Cygwin packages as of right now (2017-03-12).
Using latest nx-libs 3.6.x as of right now (dc8477f76372e685eb999a71afb4e5ec7e3176f1)

```
g++ -c  -O3 -fno-rtti -fno-exceptions -I../nx-X11/exports/include -Wmissing-declarations -DIN_ADDR_T=in_addr_t -DVERSION=\"3.5.99.4\"  -DPACKAGE_NAME=\"\" -DPACKAGE_TARNAME=\"\" -DPACKAGE_VERSION=\"\" -DPACKAGE_STRING=\"\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -Wall -Wpointer-arith   Loop.cpp
Loop.cpp:108:8: error: redefinition of ‘struct sockaddr_un’
 struct sockaddr_un
        ^
In file included from Loop.cpp:47:0:
/usr/include/sys/un.h:18:8: error: previous definition of ‘struct sockaddr_un’
 struct sockaddr_un {
        ^
make: *** [Makefile:100: Loop.o] Error 1
```

[Full Log](http://pastebin.com/Ku31i2cu)

**Comment by sunweaver** [comment]
Hi Mike,

On  So 12 Mär 2017 17:19:21 CET, Mike DePaulo wrote:

> Using latest 32-bit Cygwin packages as of right now (2017-03-12).
> Using latest nx-libs 3.6.x as of right now
>
> `
> g++ -c  -O3 -fno-rtti -fno-exceptions -I../nx-X11/exports/include  
> -Wmissing-declarations -DIN_ADDR_T=in_addr_t -DVERSION=\"3.5.99.4\"   
> -DPACKAGE_NAME=\"\" -DPACKAGE_TARNAME=\"\" -DPACKAGE_VERSION=\"\"  
> -DPACKAGE_STRING=\"\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\"  
> -Wall -Wpointer-arith   Loop.cpp
> Loop.cpp:108:8: error: redefinition of ‘struct sockaddr_un’
>  struct sockaddr_un
>         ^
> In file included from Loop.cpp:47:0:
> /usr/include/sys/un.h:18:8: error: previous definition of ‘struct  
> sockaddr_un’
>  struct sockaddr_un {
>         ^
> make: *** [Makefile:100: Loop.o] Error 1
> `
>
> [Full Log](http://pastebin.com/Ku31i2cu)

The question is, how Unix domain socket support in C(++) should be  
handled in Cygwin. Should we actually disable the nxcomp socketeering  
support on Cygwin32? Or is there a way to have Unix file socket  
support on Cygwin platforms.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by mikedep333** [comment]
@sunweaver Cygwin supports them. Only Cygwin binaries can use them, so VcXsrv can not.

**Comment by sunweaver** [comment]
Hi Mike

On  Di 14 Mär 2017 14:26:15 CET, Mike DePaulo wrote:

> @sunweaver Cygwin supports them. Only Cygwin binaries can use them,  
> so VcXsrv can not.

so we should actually get them working, because the Unix socketeering  
stuff is not for communication with the Xserver but with e.g. SSH  
offering tunnels that have socket file endpoints client and server side.

Any idea for a patch?

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by sunweaver** [comment]
Resolved by merge of #416
